File: LinkedList/DeleteDuplicateNode.js
//Delete duplicate-value nodes from a sorted linked list

function Node(data, next) {
    this.data = data;
    this.next = next;
}

function LinkedList() {
    this.currnode = null;
}

LinkedList.prototype = {
    add: function(data) {
        var newNode = new Node(data, this.currnode);
        this.currnode = newNode;
        return this.currnode;
    },
    print: function(head) {
        while (head) {
            console.log(head.data);
            head = head.next;
        }
    },
    deleteDuplicate: function(head) {
        if (head) {
            var test = head;
            while (head.next) {
                if (head.data == head.next.data) {
                    head.next = head.next.next;
                } else {
                    head = head.next;
                }
            }
            return test;
        }

    }
}

var li = new LinkedList();
li.add(2);
li.add(3);
var head1 = li.add(3);
li.print(head1);
console.log('________________')
var head2 = li.deleteDuplicate(head1);
li.print(head2);

File: LinkedList/DeleteDuplicateNode.py
# Delete duplicate-value nodes from a sorted linked list

class Node:
    def __init__(self,data,next):
        self.data = data
        self.next = next

class LinkedList:
    def __init__(self):
        self.currnode = None

    def add(self,data):
        newNode = Node(data,self.currnode)
        self.currnode = newNode
        return self.currnode
    
    def printList(self,head):
        while head:
            print head.data
            head = head.next
    
    def RemoveDuplicates(self,head):
        if head:
            node = head
            while head.next:
                if head.data == head.next.data:
                    head.next = head.next.next
                else:
                    head = head.next
        return node      
    
if __name__ == "__main__":
    li = LinkedList()
    li.add(1)
    li.add(1)
    head1 = li.add(2)
    li.printList(head1)
    print "__________"
    head2 = li.RemoveDuplicates(head1)
    li.printList(head2)